src/mon/PGMonitor.cc

```diff
@@ -935,23 +935,15 @@ void PGMonitor::check_osd_map(epoch_t epoch)
     }
   }
 
-  bool propose = false;
-  if (pg_map.last_osdmap_epoch < epoch) {
-    pending_inc.osdmap_epoch = epoch;
-    propose = true;
-  }
-
-  if (map_pg_creates())
-    propose = true;
-  if (register_new_pgs())
-    propose = true;
+  assert(pg_map.last_osdmap_epoch < epoch);
+  pending_inc.osdmap_epoch = epoch;
+  (void)map_pg_creates();
```

Comment:
`check_osd_map()` is the only caller of these functions. maybe we can change them so that they return `void` instead?